spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java

```diff
@@ -94,11 +141,15 @@ public static void startMetastoreAndSpark() {
 
     spark = SparkSession.builder()
         .master("local[2]")
+        .config("spark.appStateStore.asyncTracking.enable", false)
```

Comment:
The default is true, which means that when the `SQLAppStatusListener` receives a `SparkListenerSQLExecutionEnd` event, it asks its `ElementTrackingStore` to asynchronously aggregate the metrics. This async call sometimes results in the metric we need not being available. To make the tests more predictable, ensure that a synchronous call to aggregate metrics is used.